builtin/credential/cert/backend.go

```diff
@@ -200,6 +204,7 @@ func (b *backend) flushTrustedCache() {
 	if b.trustedCache != nil { // defensive
 		b.trustedCache.Purge()
 	}
+	b.trustedCacheFull.Store(nil)
```

Comment:
Should this be protected by the lock as well?

Reply:
Hmm, yeah, but it can't use the LockEntry pattern to do it since it'd have to be the right lock.

Reply:
Actually, no, the failure here would be some callers might get a cache hit as we start to purge, which is still possible if they complete for lock acquisition.  If they do a Get after purge that's fine too.

Reply:
Yup, I see it now, thanks!